app/src/main/java/ch/epfl/polychef/recipe/RecipeStorage.java

```diff
@@ -89,9 +91,26 @@ public void updateRecipe(@NonNull Recipe recipe) {
      * @param recipe the {@code Recipe to add}
      */
     public void addRecipe(@NonNull Recipe recipe) {
-        DatabaseReference ref = getFirebaseDatabase().getReference(DB_NAME).push();
-        recipe.setKey(ref.getKey());
-        ref.setValue(recipe);
+        
+        CallHandler<Recipe> callHandler =new CallHandler<Recipe>(){
+            @Override
+            public void onSuccess(Recipe data) {//if the recipe exist, then override it
+                if(data!=null) {
+                    getFirebaseDatabase()
+                            .getReference(RecipeStorage.DB_NAME + "/" + data.getKey())
+                            .setValue(recipe);
+                }
+            }
+
+            @Override
+            public void onFailure() {//if the recipe does not exist, add it
+                DatabaseReference ref = getFirebaseDatabase().getReference(DB_NAME).push();
+                recipe.setKey(ref.getKey());
+                ref.setValue(recipe);
+            }
+        };
+
+        readRecipeFromUuid(recipe.getRecipeUuid(),callHandler);
```

Comment:
Works fine but I feel it would have been more straight-forward to keep `addRecipe` for new recipes and `updateRecipe` for modifying old ones (since we usually know if the recipe already exists)

Reply:
I will try to make it work without that method, that would add increase coverage.
At the beginning I created that method because the PostRecipeFragment was not aware whether the recipe was new or a modified copy, wich is not the case anymore now ! I think I can enhance this, thanks